docs/TuxArchitecture.md

```diff
@@ -0,0 +1,186 @@
+
+##<a id="Implementation"></a>Implementation [#](#Implementation)
+TuxedoJS provides two major innovations for defining store dependencies:
+
+1. It allows you to write all of your store dependencies in a single location so only one file is responsible for managing your app architecture
+2. It allows you to describe your store dependencies in terms of the inputs to and outputs from each of your stores
+
+###<a name="Requiring-Architecture"></a>Requiring Architecture [#](#Requiring-Architecture)
+The Architecture interface is exposed via:
+
+```
+    var Architecture = require('tux/Architecture');
+```
+
+###<a id="Architecting-Store-Dependencies"></a>Architecting Store Dependencies [#](#Architecting-Store-Dependencies)
+Let's take a look at an example. Imagine we want to describe the following network of store dependencies where each store is dependent on the store it points to: [#](#Architecting-Store-Dependencies-Example)
+<a id="Architecting-Store-Dependencies-Example"></a>
+
+```
+    userStore <---privateMessageStore
+        ^               ^
+        |               |
+    roomStore <---------|
+        ^               |
+        |               |
+    threadStore <---unreadStore
+        ^               |
+        |               |
+    messageStore <------|
+```
+
+In Flux we would need to write these dependencies spread out across 6 different files. In TuxedoJS, we can do the following:
+
+```
+    var userStore = require('../stores/userStore');
+    var roomStore = require('../stores/roomStore');
+    var threadStore = require('../stores/threadStore');
+    var messageStore = require('../stores/messageStore');
+    var privateMessageStore = require('../stores/privateMessageStore');
+    var unreadStore = require('../stores/unreadStore');
+
+    var architect = require('tux/Architecture').architect;
+
+    architect(userStore).itOutputs('users');
+    architect(roomStore).itNeeds('users').itOutputs('rooms', 'unreadCount');
+    architect(threadStore).itNeeds('rooms').itOutputs('threads', 'unreadCount');
+    architect(messageStore).itNeeds('threads').itOutputs('messages', 'unreadCount');
+    architect(privateMessageStore).itNeeds('users').itOutputs('privateMessages', 'unreadCount');
+    architect(unreadStore).itNeeds('unreadCount');
+```
+
+TuxArchitecture also supports arrays of inputs for `itNeeds` and `itOutputs` as well as the ability to chain invocations of these functions via the `and` method (it copies the last method invoked). For example, we could rewrite `architect(roomStore)` as:
+
+```
+    architect(roomStore).itNeeds('users').itOutputs(['rooms', 'unreadCount']); //OR
+    architect(roomStore).itNeeds('users').itOutputs('rooms').and('unreadCount');
+```
+
+Or if we decide that the `unreadStore` was also dependent on the `userStore` we could write:
```

Comment:
Recommend against starting sentence with 'or'. Either 'decide' needs to be past tense or 'was' needs to be present tense.